pkg/migration-tool/utils/utils.go

```diff
@@ -0,0 +1,92 @@
+package utils
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+const (
+	LabelJob = "job"
+	Megabyte = 1024 * 1024
+)
+
+// authStore is used while creating a client. This is done to avoid complexity in the writer and reader code, thereby
+// keeping them independent of the auth.
+var authStore sync.Map
+
+// Auth defines the authentication for prom-migrator.
+type Auth struct {
+	Username    string
+	Password    string
+	BearerToken string
+}
+
+// Convert converts the auth credentials to HTTP client compatible format.
+func (a *Auth) ToHTTPClientConfig() config.HTTPClientConfig {
+	return config.HTTPClientConfig{
+		BasicAuth: &config.BasicAuth{
+			Username: a.Username,
+			Password: config.Secret(a.Password),
+		},
+		BearerToken: config.Secret(a.BearerToken),
+	}
+}
+
+// SetAuthStore sets the authStore map.
+func SetAuthStore(clientType uint, clientConfig config.HTTPClientConfig) error {
+	if _, ok := authStore.Load(clientType); ok {
+		return fmt.Errorf("auth store is read-only. Attempting to change pre-existing key: %d", clientType)
+	}
+	authStore.Store(clientType, clientConfig)
+	return nil
+}
+
+// CreatePrombRequest creates a new promb query based on the matchers.
+func CreatePrombQuery(mint, maxt int64, matchers []*labels.Matcher) (*prompb.Query, error) {
+	ms, err := toLabelMatchers(matchers)
+	if err != nil {
+		return nil, fmt.Errorf("fetch: %w", err)
+	}
+	return &prompb.Query{
+		StartTimestampMs: mint,
+		EndTimestampMs:   maxt,
```

Comment:
if we are sticking to the idea that maxt is exclusive  then I think this should be 
```suggestion
		EndTimestampMs:   maxt-1,
```

Reply:
I am not sure, as the read-storage by itself will skip the maxt. If we do this, then the time to be considered as end will actually be `maxt-2` which is wrong.

Reply:
Ah, i get the point. yes, it should be that way.